velox/common/encode/Base64.cpp

```diff
@@ -186,27 +184,31 @@ size_t Base64::calculateEncodedSize(size_t inputSize, bool withPadding) {
 }
 
 // static
-void Base64::encode(const char* input, size_t inputSize, char* output) {
-  encodeImpl(
+Status Base64::encode(const char* input, size_t inputSize, char* output) {
+  return encodeImpl(
       folly::StringPiece(input, inputSize), kBase64Charset, true, output);
 }
 
 // static
-void Base64::encodeUrl(const char* input, size_t inputSize, char* output) {
-  encodeImpl(
-      folly::StringPiece(input, inputSize), kBase64UrlCharset, true, output);
+Status
+Base64::encodeUrl(const char* input, size_t inputSize, char* outputBuffer) {
+  return encodeImpl(
+      folly::StringPiece(input, inputSize),
+      kBase64UrlCharset,
+      true,
+      outputBuffer);
 }
 
 // static
 template <class T>
-void Base64::encodeImpl(
+Status Base64::encodeImpl(
```

Comment:
Does encodeImpl throw? If not why use Status?

Reply:
`base64ReverseLookup` call is being made from `encodeImpl`, Can you have a look into the updated code.

Reply:
Isn't `base64ReverseLookup ` being called from `decodeImpl`?
I don't see any function call being made inside `encodeImpl`.